operator/internal/controller/auto/update_controller.go

```diff
@@ -168,6 +192,36 @@ func (r *UpdateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 }
 
+func isWorkspaceReady(ws *autov1alpha1.Workspace) bool {
+	if ws == nil || ws.Generation != ws.Status.ObservedGeneration {
+		return false
+	}
+	return meta.IsStatusConditionTrue(ws.Status.Conditions, autov1alpha1.WorkspaceReady)
+}
+
+type workspaceReadyPredicate struct {
+	predicate.Funcs
+}
+
+func (workspaceReadyPredicate) Create(e event.CreateEvent) bool {
+	return isWorkspaceReady(e.Object.(*autov1alpha1.Workspace))
+}
+
+func (workspaceReadyPredicate) Delete(e event.DeleteEvent) bool {
+	return false
+}
+
+func (workspaceReadyPredicate) Update(e event.UpdateEvent) bool {
+	if e.ObjectOld == nil || e.ObjectNew == nil {
+		return false
+	}
+	return !isWorkspaceReady(e.ObjectOld.(*autov1alpha1.Workspace)) && isWorkspaceReady(e.ObjectNew.(*autov1alpha1.Workspace))
```

Comment:
Are we guaranteed to see this edge event?

Reply:
If the informer becomes disconnected, it does a 're-sync' by sending a `Create` event for all existing objects. So, the Create event also checks whether the object is ready.